client/authority-discovery/src/service.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::ServicetoWorkerMsg;
+
+use futures::channel::{mpsc, oneshot};
+use futures::SinkExt;
+
+use sc_network::{Multiaddr, PeerId};
+use sp_authority_discovery::AuthorityId;
+
+/// Service to interact with the [`Worker`].
+#[derive(Clone)]
+pub struct Service {
+	to_worker: mpsc::Sender<ServicetoWorkerMsg>,
+}
+
+/// A [`Service`] allows to interact with a [`Worker`], e.g. by querying the
+/// [`Worker`]'s local address cache for a given [`AuthorityId`].
+impl Service {
+	pub(crate) fn new(to_worker: mpsc::Sender<ServicetoWorkerMsg>) -> Self {
+		Self {
+			to_worker,
+		}
+	}
+
+	/// Get the addresses for the given [`AuthorityId`] from the local address cache.
+	pub async fn get_addresses_by_authority_id(&mut self, authority: AuthorityId) -> Option<Vec<Multiaddr>> {
```

Comment:
Could you document that the addresses are always of the form `.../p2p/...`?
And also document that `None` means that the authority isn't in the cache.